internal/agent/device/config/controller.go

```diff
@@ -18,6 +19,7 @@ type Controller struct {
 	hookManager  hook.Manager
 	deviceWriter fileio.Writer
 	log          *log.PrefixLogger
+	initialized  atomic.Bool
```

Comment:
we shouldn't need atomic bool here the reconciliation loops are single threaded right?

Reply:
If we move to bootstrap then it is not needed